modules/aws/ssm.go

```diff
@@ -68,3 +73,127 @@ func NewSsmClientE(t testing.TestingT, region string) (*ssm.SSM, error) {
 
 	return ssm.New(sess), nil
 }
+
+// WaitForSsmInstanceE waits until the instance get registered to the SSM inventory.
+func WaitForSsmInstanceE(t testing.TestingT, awsRegion, instanceID string, timeout time.Duration) error {
+	timeBetweenRetries := 2 * time.Second
+	maxRetries := int(timeout.Seconds() / timeBetweenRetries.Seconds())
+	description := fmt.Sprintf("Waiting for %s to appear in the SSM inventory", instanceID)
+
+	key := aws.String("AWS:InstanceInformation.InstanceId")
+	filterType := aws.String("Equal")
+	values := aws.StringSlice([]string{instanceID})
+	_, err := retry.DoWithRetryE(t, description, maxRetries, timeBetweenRetries, func() (string, error) {
+		client := NewSsmClient(t, awsRegion)
+		req, resp := client.GetInventoryRequest(&ssm.GetInventoryInput{
```

Comment:
NIT: probably makes sense to declare `input` as a variable outside the retry loop, with all the filter params defined in-line, rather than defining the filter params outside with separate named variables and the `input` inside.